chain/jsonrpc/src/lib.rs

```diff
@@ -18,17 +18,15 @@ use near_client::{
     GetStateChangesInBlock, GetValidatorInfo, GetValidatorOrdered, Query, Status, TxStatus,
     TxStatusError, ViewClientActor,
 };
-pub use near_jsonrpc_client as client;
-use near_jsonrpc_primitives::errors::RpcError;
-use near_jsonrpc_primitives::message::{Message, Request};
-#[cfg(feature = "adversarial")]
-use near_jsonrpc_primitives::types::adversarial::SetAdvOptionsRequest;
 #[cfg(feature = "adversarial")]
+use near_jsonrpc_adversarial_primitives::SetAdvOptionsRequest;
 #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
-use near_jsonrpc_primitives::types::adversarial::SetRoutingTableRequest;
-#[cfg(feature = "adversarial")]
+use near_jsonrpc_adversarial_primitives::SetRoutingTableRequest;
 #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
-use near_jsonrpc_primitives::types::adversarial::StartRoutingTableSyncRequest;
+use near_jsonrpc_adversarial_primitives::StartRoutingTableSyncRequest;
```

Comment:
should this be under `adversarial` as well @pmnoxx ?

Reply:
It's covered here https://github.com/near/nearcore/pull/4912/files#diff-2b29afffc8afa7eed1365ace3137ccec9802b75eb3b5dd35dc4fd4d388a4c099R45

Reply:
This dependency is strange. I suggest that we decouple them

Reply:
Those functions are used for testing `protocol_feature_routing_exchange_algorithm`.
Therefore you need `adversarial` for testing, and `protocol_feature_routing_exchange_algorithm` for the feature itself.